Tests/iaas/flavor-naming/check-yaml.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+"""Check flavor names in the YAML files used for scs-0103-v1 with respect to scs-0100-vX.
+
+Also check that the flavor properties encoded in the YAML match those encoded in the name.
+"""
+import importlib
+import os
+import os.path
+import sys
+
+import yaml
+
+fnmck = importlib.import_module("flavor-name-check")
+fnmck.disallow_old = True
+
+
+REQUIRED_FIELDS = ['name-v1', 'name-v2', 'name', 'cpus', 'ram', 'cpu-type']
+DEFAULTS = {'disk0-type': 'network'}
+CPUTYPE_KEY = {'L': 'crowded-core', 'V': 'shared-core', 'T': 'dedicated-thread', 'C': 'dedicated-core'}
+DISKTYPE_KEY = {'n': 'network', 'h': 'hdd', 's': 'ssd', 'p': 'nvme'}
+
+
+class Undefined:
+    def __repr__(self):
+        return 'undefined'
+
+
+class Checker:
+    def __init__(self):
+        self.errors = 0
+
+    def emit(self, s):
+        print(f"ERROR: {s}", file=sys.stderr)
+        self.errors += 1
+
+    def check_spec(self, flavor_spec):
+        missing = [key for key in REQUIRED_FIELDS if key not in flavor_spec]
+        if missing:
+            self.emit(f"flavor spec missing keys {', '.join(missing)}: {flavor_spec}")
+            return
+        name = flavor_spec['name']
+        name_v2 = flavor_spec['name-v2']
+        parsed = fnmck.parsename(name_v2)
+        if not parsed:
+            self.emit(f"flavor {name}: name-v2 '{name_v2}' could not be parsed")
+            return
+        cpu, disk, hype, hvirt, cpubrand, gpu, ibd = parsed
+        undefined = Undefined()
+        expected = {
+            'cpus': cpu.cpus,
+            'cpu-type': CPUTYPE_KEY[cpu.cputype],
+            'ram': cpu.ram,
+            'name-v1': fnmck.new_to_old(name_v2),
+            'disk': undefined,
+        }
+        if disk.parsed:
+            if disk.nrdisks != 1:
+                self.emit(f"flavor '{name}': name-v2 using multiple disks")
+            expected['disk'] = disk.disksize
+            expected['disk0-type'] = DISKTYPE_KEY[disk.disktype or 'n']
+        for key, exp_val in expected.items():
+            val = flavor_spec.get(key, DEFAULTS.get(key, undefined))
+            if val != exp_val:
+                self.emit(
+                    f"flavor '{name}': field '{key}' contradicting name-v2 '{name_v2}'; "
+                    f"found {val!r}, expected {exp_val!r}"
+                )
+
+
+def main(argv):
+    if len(argv) != 2:
+        raise RuntimeError("must specify exactly one argument, PATH to the yaml file")
+    yaml_path = argv[1]
+    yaml_files = sorted([
+        fn
```

Comment:
nit/opinion: I prefer longer, readable variable names, like `filename` in this case

Reply:
I don't care either way, but I guess it would be better for some people, so I will use `filename`